clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapServerStore.java

```diff
@@ -49,12 +49,13 @@
     }
   }
 
-  public OffHeapServerStore(ResourcePageSource source, KeySegmentMapper mapper) {
+  public OffHeapServerStore(ResourcePageSource source, KeySegmentMapper mapper, boolean isShared) {
     this.mapper = mapper;
     segments = new ArrayList<>(mapper.getSegments());
-    long maxSize = getMaxSize(source.getPool().getSize());
+    int minPageSize = KILOBYTES.toBytes(4);
+    int maxPageSize = isShared ? minPageSize : (int) KILOBYTES.toBytes(getMaxSize(source.getPool().getSize()));
```

Comment:
Like @chrisdennis, I don't know why this would solve the problem. As usual, an explanation around the code (in the code) would be nice. Same thing in the test to explain what it does.